From 302071a07087bad22078fcc339684875425f26fe Mon Sep 17 00:00:00 2001
From: Harikrushna Parmar <harikrushna.parmar@nxp.com>
Date: Mon, 30 Sep 2024 13:41:41 +0530
Subject: [PATCH 5/6] MGG-1448 [#ccc] svgpathtools: bugfix: Handle polyline
 path correctly

if path string or is_polygon flag is enableed then append path given in input
if path string and is_polygon flag is enable then append 'z' path

Signed-off-by: Harikrushna Parmar <harikrushna.parmar@nxp.com>
---
 svgpathtools/svg_to_paths.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/svgpathtools/svg_to_paths.py b/svgpathtools/svg_to_paths.py
index 0d5f47e..824d062 100644
--- a/svgpathtools/svg_to_paths.py
+++ b/svgpathtools/svg_to_paths.py
@@ -78,7 +78,7 @@ def polyline2pathd(polyline, is_polygon=False):
     # The `parse_path` call ignores redundant 'z' (closure) commands
     # e.g. `parse_path('M0 0L100 100Z') == parse_path('M0 0L100 100L0 0Z')`
     # This check ensures that an n-point polygon is converted to an n-Line path.
-    if is_polygon and closed:
+    if is_polygon or closed:
         points.append(points[0])
 
     d = 'M ' + ' L '.join('{0} {1}'.format(x,y) for x,y in points)
-- 
2.34.1

